src/scripts/api.ts

```diff
@@ -0,0 +1,241 @@
+            promises.push(item);
+          }
+        }
+      } else {
+        const item = getDataForSingleYearAndState(selectedOptions, state, years[0]);
+        promises.push(item);
+      }
+    }
+  } else {
+    if (countySelected) {
+      const allCounties = countyData[states[0]];
+      for (const county of allCounties) {
+        const item = getDataForSingleYearAndState(selectedOptions, county, years[0], states[0]);
+        promises.push(item);
+      }
+    } else {
+      const item = getDataForSingleYearAndState(selectedOptions, states[0], years[0]);
+      promises.push(item);
+    }
+  }
+  return await Promise.all(promises);
+};
+
+const getDataForSingleYearAndState = async (selectedOptions: IStateOptions, countyOrState: string, year: string, state?: string) => {
+  const {geographicLevel, states, years, cropUnits, ...subOptions} = selectedOptions;
+
+  let item: any = {
+    [geographicLevel]: countyOrState,
+    "Year": year,
+  };
+
+  for (const key in subOptions) {
+    const value = subOptions[key as keyof typeof subOptions];
+    if (value && Array.isArray(value)) {
+      for (const attribute of value) {
+        const queryParams = getQueryParams(attribute);
+        const yearAvailable = queryParams?.years[geographicLevel].includes(year);
+        const isMultiStateRegion = queryParams?.geographicAreas[0] === "REGION : MULTI-STATE";
+        const geoLevel = isMultiStateRegion ? "REGION : MULTI-STATE" : geographicLevel;
+        if (yearAvailable) {
+          let location = countyOrState;
+          if (isMultiStateRegion) {
+            const itemToCheck = state ? state : countyOrState;
+            const regData = multiRegions.find((region) => region.States.includes(itemToCheck));
+            location = regData?.Region ? regData.Region : countyOrState;
+          }
+          const params: IGetAttrDataParams = {attribute, geographicLevel: geoLevel, location, year, cropUnits};
+          if (geoLevel === "County") {
+            params.state = state;
+          }
+          const attrData = await getAttrData(params);
+          item = {...item, ...attrData};
+        }
+      }
+    }
+  }
+
+  return item;
+};
+
+const getAttrData = async (params: IGetAttrDataParams) => {
+  const {attribute, geographicLevel, location, year, cropUnits, state} = params;
+  const reqParams: IRequestParams = {attribute, geographicLevel, location, year, state};
+  if (cropOptions.options.includes(attribute) && cropUnits) {
+    reqParams.cropCategory = cropUnits as keyof ICropDataItem;
+  }
+  const req = createRequest(reqParams);
+  const res = await fetchDataWithRetry(req);
+  const values: any = {};
+  if (res) {
+    const {data} = res;
+    data.map((dataItem: any) => {
+       const codapColumnName = attrToCODAPColumnName[dataItem.short_desc].attributeNameInCodapTable;
+       return values[codapColumnName] = dataItem.Value;
+    });
+  } else {
+    console.log("error");
```

Comment:
add an eslint exception for the console.log, and may be a more descriptive error message ("Error: Did not recesive response from query").